src/CountUp.tsx

```diff
@@ -108,18 +121,19 @@ class CountUp extends Component {
     // end value.
     if (end !== prevProps.end) {
       if (!preserveValue) {
-        this.instance.reset();
+        this.instance?.reset();
       }
-      this.instance.update(end);
+      this.instance?.update(end);
     }
   }
 
   componentWillUnmount() {
     if (this.timeoutId) {
       clearTimeout(this.timeoutId);
     }
-    if (this.instance.target) {
-      this.instance.reset();
+    // instance.target is incorrectly marked private by typescript
+    if ((this.instance as any).target) {
+      this.instance?.reset();
     }
```

Comment:
I think that we should simplify this to:
```js
this.instance?.reset();
```

and we can skip creation of **CountUp** instance in `createInstance`:
```js
  createInstance = () => {
    if (typeof this.props.children === 'function') {
      // Warn when user didn't use containerRef at all
      warning(
        this.containerRef.current &&
          (this.containerRef.current instanceof HTMLElement ||
            this.containerRef.current instanceof SVGTextElement ||
            this.containerRef.current instanceof SVGTSpanElement),
        `Couldn't find attached element to hook the CountUp instance into! Try to attach "containerRef" from the render prop to a an HTMLElement, eg. <span ref={containerRef} />.`,
      );
      return;
    }
    return createCountUpInstance(this.containerRef.current, this.props);
  };
```

Reply:
This PR is meant to be a types-only change and while I'm sure that check could be simpler, I'd recommend non-ts changes be added to my other PR moving to a hook-first approach.

Reply:
Also, the component CountUp must actually initialize the instance on mount for backwards compatibility - the tests all test for CountUp having 0 (props.start) in the component instead of an empty container.

Reply:
I moved the this.instance check to the right spot (before checking this.instance.target). This createInstance function is actually for creating the CountUpJs instance, so I'm not sure where it would be created if not here.